src/util/irep_serialization.cpp

```diff
@@ -324,11 +324,15 @@ Function: irep_serializationt::read_gb_string
 
 irep_idt irep_serializationt::read_gb_string(std::istream &in)
 {
-  char c;
   size_t length=0;
 
-  while((c = in.get()) != 0)
+  while(char c = in.get())
   {
+    if(!in.good())
+    {
+      throw irep_serialization_errort("gb string is not null-terminated");
```

Comment:
Not until #911 is merged, or you change all other 1583 uses of `throw` without a user-defined type.